### PR TITLE
docs(web-demo): add consumer config matrix for retail and UI

### PR DESCRIPTION
**Pull request by mmcshinsky-bitgo**
## Summary
- Expand \`modules/web-demo/README.md\` with a consumer config matrix mapping planned \`CONSUMER_PROFILE\` setups to bitgo-retail and bitgo-ui bundling patterns
- Document current demo routes and which production problems each route illustrates
- Extend \`commitlint.config.js\` with \`WEB-\` as an issue prefix for side-branch work
- Update \`.github/workflows/ci.yml\` to run a new \`code-quality (scripts)\` check against \`scripts/tests/\`
- Add \`CLAUDE.md\` note on the junk-drawer \`WEB-000\` reference pattern
- Refactor \`scripts/prepare-commit-msg.js\`: extract pure \`extractTicket\` / \`appendTicketFooter\` functions; bail when \`issuePrefixes\` is empty to avoid matching bare-digit branches; fix \`commitRegex.lastIndex\` leakage across repeated \`.test()\` calls
- Add \`scripts/tests/prepare-commit-msg.test.js\` with 12 unit tests covering extraction, footer append, dedup, and regex-safety edge cases

## Test plan
- [x] \`yarn run test:scripts\` — 12 passing
- [x] Commitlint passes on all commits in range
- [ ] Review matrix accuracy for retail WASM aliases and auth flows (manual verify before merge)

No ticket — local side improvement.

### CODE REVIEW

File: scripts/prepare-commit-msg.js [hunk 1]
```diff
@@ -5,21 +5,35 @@ const childProcess = require('child_process');
 const fs = require('fs');
 const exec = promisify(childProcess.exec);
 
-// ex WP-1234
-const branchRegex = /([A-Z]+-)(\d+)/;
+// Build regex from commitlint's known issuePrefixes so arbitrary branch segments
```
Comment (OttoAllmendinger) [inline scripts/prepare-commit-msg.js:8]: ironically, a script that deserves tests!

**Review by OttoAllmendinger — DISMISSED** [review]

**Review by OttoAllmendinger — APPROVED** [review]

**Comment by mmcshinsky-bitgo** [comment]
## Test Plan

### 1. Setup

```bash
git fetch origin
git checkout <PR branch>
```

### 2. Unit tests — `prepare-commit-msg.js`

```bash
yarn run test:scripts
```

**Expected:** 12 tests pass covering ticket extraction, footer append, dedup (case-insensitive), null-ticket skip, bare-digit branch no-match, and `commitRegex.lastIndex` safety.

### 3. Commitlint — `WEB-` prefix

```bash
echo "chore(root): test WEB- prefix\n\nWEB-000" | yarn commitlint
```

**Expected:** exits 0.

### 4. CI — `code-quality (scripts)` job

Verify the new job appears and passes in the PR's CI checks (`.github/workflows/ci.yml`).

### 5. Hook end-to-end

```bash
git checkout -b WEB-123-scratch
# make a trivial change, stage it
git commit -m "chore(test): scratch commit"
git log -1 --format="%B"
```

**Expected:** commit message ends with `TICKET: WEB-123`; no duplicate footer on `--amend`.

### 6. Manual — `modules/web-demo/README.md`

Review the consumer config matrix: confirm `CONSUMER_PROFILE` values map correctly to `bitgo-retail`/`bitgo-ui` bundling patterns, WASM aliases, and auth flows.

**Review by Samyakk123 — APPROVED** [review]

**Review by zahin-mohammad — APPROVED** [review]
